### PR TITLE
feat(sqlalchemy): extend SQLAlchemy Plugin to include Alembic

### PR DESCRIPTION
**Pull request by cofin**
### Pull Request Checklist

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

### Description
[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

-

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

-

### CODE REVIEW

File: litestar/contrib/sqlalchemy/plugins/init/plugin.py [hunk 1]
```diff
@@ -2,19 +2,22 @@
 
 from typing import TYPE_CHECKING
 
+from litestar.contrib.sqlalchemy.cli import database_group
```
Comment (provinzkraut) [inline litestar/contrib/sqlalchemy/plugins/init/plugin.py:5]: You'll need to put this import inside the `on_cli_init` method, else users will get an `ImportError` when they don't have the CLI dependencies installed

File: litestar/contrib/sqlalchemy/alembic/commands.py
Comment (provinzkraut) [inline litestar/contrib/sqlalchemy/alembic/commands.py:1]: What is the purpose of these commands being async?
They don't seem to be doing anything asynchronous

Comment (cofin) [reply litestar/contrib/sqlalchemy/alembic/commands.py:1]: I'm not sure exactly.  My original intent was to try and create a single set of functions that work for sync and async.  I'll remove the async if there's no reason to have it.

Comment (provinzkraut) [reply litestar/contrib/sqlalchemy/alembic/commands.py:1]: I've just never seen async migration functions :shrug: 

I guess it makes some sort of sense if you're calling them from an async context, but then you'd have to wrap the alembic commands themselves in something like `anyio.to_thread.run_sync` to make them non-blocking.

**Comment by provinzkraut** [comment]
Maybe stamp would be another good command to add? I find that quite useful for development sometimes.

File: litestar/contrib/sqlalchemy/alembic/commands.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    plugin = app.plugins.get(SQLAlchemyInitPlugin)
+    alembic_cfg = get_alembic_command_config(config=plugin._config)
+    migration_command.history(
+        config=alembic_cfg, rev_range=rev_range, verbose=verbose, indicate_current=indicate_current
+    )
+
+
+def merge(
+    app: Litestar,
+    revisions: str,
```
Comment (provinzkraut) [inline litestar/contrib/sqlalchemy/alembic/commands.py:157]: Is this correct? Shouldn't this be `tuple[str, str]` or something like that?

**Comment by provinzkraut** [comment]
> Maybe stamp would be another good command to add? I find that quite useful for development sometimes.

Never mind, it's already in there (=

File: litestar/contrib/sqlalchemy/alembic/commands.py (hunk 2 above)
Comment (cofin) [reply litestar/contrib/sqlalchemy/alembic/commands.py:157]: I'll take a look at this one.  I started by making wrappers around each alembic command and I'm slowly adding all the commands to the CLI.

**Comment by cofin** [comment]
@provinzkraut I may need your input on this one.  I need a way to inject the database URL into the `env.py`.  The method you see that i've implemented results in the value of the DB URL being `None`.  This is one of the last pieces to solve to make it run.

**Comment by provinzkraut** [comment]
> @provinzkraut I may need your input on this one. I need a way to inject the database URL into the `env.py`. The method you see that i've implemented results in the value of the DB URL being `None`. This is one of the last pieces to solve to make it run.

Do you mean this?
https://github.com/litestar-org/litestar/blob/94afae2fc4eb9dc54c60f6fbfd8453b63976a237/litestar/contrib/sqlalchemy/alembic/commands.py#L76

**Comment by cofin** [comment]
> 

Yep.  This is the section that doesn't seem to be working.  I also tried passing the engine itself and that didn't work.

**Pull request closed without merging**